pkg/mbw/monitor/monitor.go

```diff
@@ -0,0 +1,359 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/klauspost/cpuid/v2"
+
+	"github.com/kubewharf/katalyst-core/pkg/config/agent/global"
+	"github.com/kubewharf/katalyst-core/pkg/mbw/utils"
+	"github.com/kubewharf/katalyst-core/pkg/mbw/utils/pci"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+)
+
+const (
+	MBM_MONITOR_INTERVAL = 1000
+
+	MAX_NUMA_DISTANCE = 32 // this is the inter-socket distance on AMD, intel inter-socket distance < 32
+	MIN_NUMA_DISTANCE = 11 // the distance to local numa is 10 in Linux, thus 11 is the minimum inter-numa distance
+)
+
+func newSysInfo(machineInfoConfig *global.MachineInfoConfiguration) (*SysInfo, error) {
```

Comment:
Can all the static sysInfo move to metaserver KatalystMachineInfo?

Reply:
yes this is a good suggestion. will refactor to make these into KatalystMachineInfo's cpu topology,  CCD topology respectively in next commit.

Reply:
code refactored and sysinfo structure is merged into katalyst machine info